### PR TITLE
Request a new filter "woocommerce_can_restock_refunded_items"

### PR DESCRIPTION
**Pull request by kreamweb**
I need this filter before process the restock for refund items.
It can be useful like woocommerce_can_reduce_order_stock or woocommerce_can_restore_order_stock

### CODE REVIEW

File: includes/data-stores/class-wc-coupon-data-store-cpt.php [hunk 1]
```diff
@@ -224,6 +224,9 @@ public function delete( &$coupon, $args = array() ) {
 	 * @since 3.0.0
 	 */
 	private function update_post_meta( &$coupon ) {
+		
+		$this->updated_props = array();
```
Comment (rodrigoprimo) [inline includes/data-stores/class-wc-coupon-data-store-cpt.php]: Why this is needed? From the commit message, this change seems unrelated to this PR. If there is a separate you that is fixed by this, could you please open a separate PR?

File: includes/wc-order-functions.php [hunk 2]
```diff
@@ -693,7 +693,7 @@ function wc_restock_refunded_items( $order, $refunded_line_items ) {
 		$item_stock_reduced = $item->get_meta( '_reduced_stock', true );
 		$qty_to_refund      = $refunded_line_items[ $item_id ]['qty'];
 
-		if ( ! $item_stock_reduced || ! $qty_to_refund || ! $product || ! $product->managing_stock() ) {
+		if ( ! $item_stock_reduced || ! $qty_to_refund || ! $product || ! $product->managing_stock() || ! apply_filters('woocommerce_can_restock_refunded_items', true, $order, $refunded_line_items ) ) {
```
Comment (rodrigoprimo) [inline includes/wc-order-functions.php]: If we decide to add this filter, I imagine that passing information about the current item to the filter is relevant, no?

**Review by rodrigoprimo — CHANGES_REQUESTED** [review]
Thanks for your contribution. I left a few questions as inline comments. Besides that, could you please elaborate on why do you need this filter? I'm asking to try to understand if a new filter is the best solution or if we have another option.

File: includes/data-stores/class-wc-coupon-data-store-cpt.php (hunk 1 above)
Comment (kreamweb) [reply includes/data-stores/class-wc-coupon-data-store-cpt.php]: Yes, this is related to a new issue.
You can ignore it, I'll open a new pull request.

File: includes/wc-order-functions.php (hunk 2 above)
Comment (kreamweb) [reply includes/wc-order-functions.php]: I'll open a new request more specific, thank you

**Comment by rodrigoprimo** [comment]
Thanks!
